ui/src/plugins/dev.perfetto.ExplorePage/query_builder/query_canvas.ts

```diff
@@ -40,15 +40,14 @@ interface NodeBoxAttrs {
 class NodeBox implements m.ClassComponent<NodeBoxAttrs> {
   view({attrs}: m.CVnode<NodeBoxAttrs>) {
     const {node, isSelected, layout, onNodeSelected, onNodeDragStart} = attrs;
+    const conditionalClasses = `${isSelected ? 'selected' : ''} ${
+      !node.validate() ? 'invalid' : ''
+    }`.trim();
```

Comment:
There's a function for this!

```ts
const conditionalClasses = classNames(
  isSelected && 'selected',
  !node.validate() && 'invalid',
);
```

Also, I'd advise against using simple class names like `selected` and `invalid` - if anyone has used these classnames anywhere else in the UI they could apply to your elements and cause issues.

I'd recommend calling them something like this:

- `node-box` -> `pf-node-box`
- `selected` -> `pf-node-box--selected`
- `invalid` -> `pf-node-box--invalid`

The `pf-` prefix is to avoid running into issues when we embed perfetto.

This is the crux of BEM. It's verbose but it's precise.